packages/card/src/vaadin-card.js

```diff
@@ -46,7 +294,62 @@ class Card extends ElementMixin(ThemableMixin(PolylitMixin(LitElement))) {
 
   /** @protected */
   render() {
-    return html`<slot></slot>`;
+    return html`
+      <div part="media">
+        <slot name="media"></slot>
+      </div>
+      <div part="header">
+        <slot name="header-prefix"></slot>
+        <slot name="header">
+          <slot name="title"></slot>
+          <slot name="subtitle"></slot>
+        </slot>
+        <slot name="header-suffix"></slot>
+      </div>
+      <div part="content">
+        <slot></slot>
+      </div>
+      <div part="footer">
+        <slot name="footer"></slot>
+      </div>
+    `;
+  }
+
+  /** @private */
+  _onSlotChange() {
+    // Chrome doesn't support `:host(:has())`, so we'll recreate that with custom attributes
+    this.toggleAttribute('has-media', this.querySelector(':scope > [slot="media"]'));
+    this.toggleAttribute('has-header', this.querySelector(':scope > [slot="header"]'));
+    this.toggleAttribute(
+      'has-title',
+      this.querySelector(':scope > [slot="title"]') && !this.querySelector(':scope > [slot="header"]'),
+    );
+    this.toggleAttribute(
+      'has-subtitle',
+      this.querySelector(':scope > [slot="subtitle"]') && !this.querySelector(':scope > [slot="header"]'),
+    );
+    this.toggleAttribute('has-header-prefix', this.querySelector(':scope > [slot="header-prefix"]'));
+    this.toggleAttribute('has-header-suffix', this.querySelector(':scope > [slot="header-suffix"]'));
+    this.toggleAttribute('has-content', this.querySelector(':scope > :not([slot])'));
```

Comment:
This could be probably updated to check for the actual slotted nodes as `querySelector()` doesn't return text nodes from the default slot. I could change the logic to use `SlotObserver` helper that we have since it provides the access to the list of current nodes using `slot.assignedNodes({ flatten: true })`.